core/shared/src/main/scala/zio/Semaphore.scala

```diff
@@ -32,7 +33,7 @@ import scala.collection.immutable.{Queue => ScalaQueue}
  * If you need functionality that `Semaphore` doesnt' provide, use a
  * [[TSemaphore]] and define it in a [[zio.stm.ZSTM]] transaction.
  */
-sealed trait Semaphore extends Serializable {
+private[zio] trait Semaphore extends Serializable {
```

Comment:
```suggestion
sealed trait Semaphore extends Serializable {
```

in `object Semaphore { private[zio] abstract class SemaphoreBase extends Semaphore`